jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -50,18 +86,105 @@ class TensorFlowInferenceIT extends TestData {
    * @return List of PredictionVector
    */
   def readPredictionCSV(csvFile: String): List[PredictionVector] = {
-    return Source.fromFile(csvFile).getLines.toList.tail.flatMap(extractColumnValues)
+    Source.fromFile(csvFile).getLines.toList.tail.flatMap(extractColumnValues)
   }
 
+  def validateRankingScores(query: StringMapQueryAndPredictions,
+                     sequenceExample: SequenceExample,
+                     scores: Array[Float],
+                     numDocs: Int) = {
+    val docScores = scores.take(numDocs)
+    val maskedScores = scores.drop(numDocs)
+    docScores.foreach(
+      score => assertTrue("all docs should score non-negative", score > 0)
+    )
+    for {
+      maskedScore <- maskedScores
+      docScore <- docScores
+    } {
+      assertTrue(
+        s"docScore ($docScore) should be > masked score ($maskedScore)",
+        docScore > maskedScore
+      )
+    }
+    if (query.predictedScores != null) {
+      assertArrayEquals("scores aren't close enough: ", docScores, query.predictedScores, 1e-6f)
+    }
+
+  }
+
+  @Test
+  def testRankingSavedModelBundleWithCSVData(): Unit = {
+    val generatedBundleLocation = System.getProperty("bundleLocation")
+    def modelName = System.getProperty("runName") + "_ranking"
+    val bundlePath = generatedBundleLocation + "models/" + modelName + "/final/tfrecord"
+    val predictionPath = generatedBundleLocation + "logs/" + modelName + "/model_predictions.csv"
+    val featureConfigPath = generatedBundleLocation + "ml4ir/applications/ranking/tests/data/configs/feature_config.yaml"
+
+    evaluateRankingInferenceAccuracy(bundlePath, predictionPath, featureConfigPath)
+  }
+
+  def evaluateRankingInferenceAccuracy(bundlePath: String, predictionPath: String, featureConfigPath: String) = {
+    val allScores: Iterable[(StringMapQueryAndPredictions, SequenceExample, Array[Float], Array[Float])] = runQueriesAgainstDocs(
+      predictionPath,
+      bundlePath,
+      featureConfigPath,
+      "serving_tfrecord_protos",
+      "StatefulPartitionedCall_1"
```

Comment:
We should probably not hardcode this. @ducouloa @jakemannix ?

Reply:
This is currently hardcoded everywhere (even in the Core-app  code - I think W-8375415 is to tackle the work) - I can create an equivalent for ml4ir project